File: content-solutions/backtracking/snack-total.js
function solution(prices, money) {
  const n = prices.length;
  function go(i, total) {
    if (i === n) return total === money ? 1 : 0;
    return go(i + 1, total + prices[i]) + go(i + 1, total);
  }
  return go(0, 0);
}

File: content-solutions/backtracking/snack-total.java
class Solution {
    private int[] prices;
    private int money;

    int go(int i, int total) {
        if (i == prices.length) return total == money ? 1 : 0;
        return go(i + 1, total + prices[i]) + go(i + 1, total);
    }

    public int solution(int[] prices, int money) {
        this.prices = prices;
        this.money = money;
        return go(0, 0);
    }
}
